src/Framework/TestRunner/ChildProcessResultProcessor.php

```diff
@@ -74,6 +74,16 @@ public function process(Test $test, string $serializedProcessResult, string $std
             return;
         }
 
+        if ($childResult->expectedProcessExit !== null && $childResult->testCalledExit === true) {
+            assert($test instanceof TestCase);
+
+            $test->assertSame($childResult->expectedProcessExit, $exitCode, 'Process exit-code expectation failed');
+        } elseif ($childResult->expectedProcessExit !== null && $childResult->testCalledExit === false) {
+            $test->fail('Process expected exit() to be called but test did not call it');
+        } elseif ($childResult->expectedProcessExit === null && $childResult->testCalledExit === false) {
+            $test->fail('Process called exit() but the test did not expect it');
+        }
```

Comment:
we need something here which allows us to combine parent-process level assertions with the child process results into a overall result